tests/test_endpoints.py

```diff
@@ -78,46 +79,58 @@ def test_wait_for_status(nhsd_apim_proxy_url, status_endpoint_auth_headers):
 
 @pytest.mark.smoketest
 def test_for_connection_status(nhsd_apim_proxy_url):
-    target_server_headers = {
-        "Content-Type": "application/json",
-        "x-api-key": "lVlQRHlM4M111q8fnmLBe201HAWMNQJH16SU8Q4C",
-        "X-Correlation-ID": "df728790-43d9-4e90-ad34-b3c8268a6674",
-        "X-Request-ID": "b452ba10-6783-449d-b23e-da146ea27140",
-    }
 
     request_body = {
         "query": (
             "query PublishedCohortLibraryGetAll { PublishedCohortLibraryGetAll "
-            "{ ... on Cohorts { cohorts { urlSlug } } "
-            "... on ErrorDescription { code correlationId errorDescription } }}"
+            "{ authors,"
+            "clinicalAtRiskGroupsText,"
+            "id,"
+            "commissioner,"
+            "demographicsText,"
+            "description,"
+            "disclaimerText,"
+            "fixedDateReference,"
+            "name,"
+            "purpose,"
+            "shortName,"
+            "urlSlug,"
+            "summary } }"
         )
     }
 
     resp = requests.post(
-        f"{nhsd_apim_proxy_url}/api", headers=target_server_headers, json=request_body
+        f"{nhsd_apim_proxy_url}/api",
+        headers=Generators.generate_target_server_headers("df728790-43d9-4e90-ad34-b3c8268a6674"), json=request_body
     )
     assert resp.status_code == 200
 
 
 @pytest.mark.smoketest
 def test_for_response_headers(nhsd_apim_proxy_url):
-    target_server_headers = {
-        "Content-Type": "application/json",
-        "x-api-key": "lVlQRHlM4M111q8fnmLBe201HAWMNQJH16SU8Q4C",
-        "X-Correlation-ID": "df728790-43d9-4e90-ad34-b3c8268a6674",
-        "X-Request-ID": "b452ba10-6783-449d-b23e-da146ea27140",
-    }
 
     request_body = {
         "query": (
             "query PublishedCohortLibraryGetAll { PublishedCohortLibraryGetAll "
-            "{ ... on Cohorts { cohorts { urlSlug } } "
-            "... on ErrorDescription { code correlationId errorDescription } }}"
+            "{ authors,"
+            "clinicalAtRiskGroupsText,"
+            "id,"
+            "commissioner,"
+            "demographicsText,"
+            "description,"
+            "disclaimerText,"
+            "fixedDateReference,"
+            "name,"
+            "purpose,"
+            "shortName,"
+            "urlSlug,"
+            "summary } }"
         )
     }
 
     resp = requests.post(
-        f"{nhsd_apim_proxy_url}/api", headers=target_server_headers, json=request_body
+        f"{nhsd_apim_proxy_url}/api",
+        headers=Generators.generate_target_server_headers("df728790-43d9-4e90-ad34-b3c8268a6674"), json=request_body
```

Comment:
Could you please add some separate tests for headers as well?

Reply:
There is some header tests at the end of this file